source/Halibut.Tests/TraceLogFileLogger.cs

```diff
@@ -1,25 +1,82 @@
 ﻿using System;
+using System.Collections.Generic;
 using System.IO;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.Threading;
 
 namespace Halibut.Tests
 {
     public class TraceLogFileLogger : IDisposable
     {
-        string tempFilePath = Path.GetTempFileName();
+        readonly AsyncQueue<string> queue = new AsyncQueue<string>();
+        readonly string tempFilePath = Path.GetTempFileName();
         string testHash;
 
+        readonly TaskCompletionSource<bool> noMoreDelaysToWritingLogs;
+        readonly Task noMoreDelaysTask;
+        readonly Task writeDataToDiskTask;
+
+        public TraceLogFileLogger()
+        {
+            noMoreDelaysToWritingLogs = new TaskCompletionSource<bool>();
+            noMoreDelaysTask = noMoreDelaysToWritingLogs.Task;
+            writeDataToDiskTask = WriteDataToFile();
+        }
+
         public void SetTestHash(string testHash)
         {
             this.testHash = testHash;
         }
         
         public void WriteLine(string logMessage)
         {
-            File.AppendAllLines(tempFilePath, new[] { logMessage });
+            queue.Enqueue(logMessage);
+        } 
+        
+        async Task WriteDataToFile()
+        {
+            while (true)
+            {
+                // Don't hammer the disk, let some log message queue up before writing them.
+                if (!noMoreDelaysTask.IsCompleted)
+                {
+                    await Task.WhenAny(Task.Delay(5), noMoreDelaysTask);
+                }
+
+                var list = new List<string>();
+                list.Add(await queue.DequeueAsync());
```

Comment:
Why do we dequeue, then do a loop of TryDequeue?

I would have thought the loop would have been enough, right?

Reply:
Because we want one call the first to block until we have something, then we want to read everything we have. To then write all of that in one go.

Reply:
Ah, I see. Might be worth adding a comment about that 🙂